filesource/filesource.go

```diff
@@ -171,6 +177,32 @@ func (src Source) Check(args airbyte.CheckCmd) error {
 	})
 }
 
+func (src Source) Discover(args airbyte.DiscoverCmd) error {
+	if src.SkipSchemaInference {
+		var conn, err = newConnector(src, args.ConfigFile)
+		if err != nil {
+			return err
+		}
+
+		return airbyte.NewStdoutEncoder().Encode(airbyte.Message{
```

Comment:
I did it this way since we'll be removing schema inference entirely in the near future, and the redundancy will be resolved then. Basically going for the minimal amount of code to change that will soon be deleted. The patch here then will be removing the `if src.SkipSchemaInference` part and always returning the message with the minimal schema.